include/dlaf/permutations/general/impl.h

```diff
@@ -0,0 +1,201 @@
+//
+// Distributed Linear Algebra with Future (DLAF)
+//
+// Copyright (c) 2018-2022, ETH Zurich
+// All rights reserved.
+//
+// Please, refer to the LICENSE file in the root directory.
+// SPDX-License-Identifier: BSD-3-Clause
+//
+
+#pragma once
+
+#include "dlaf/permutations/general/api.h"
+
+#include "dlaf/blas/tile.h"
+#include "dlaf/common/index2d.h"
+#include "dlaf/lapack/gpu/lacpy.h"
+#include "dlaf/lapack/tile.h"
+#include "dlaf/matrix/copy_tile.h"
+#include "dlaf/matrix/matrix.h"
+#include "dlaf/schedulers.h"
+#include "dlaf/sender/transform.h"
+#include "dlaf/sender/when_all_lift.h"
+#include "dlaf/types.h"
+#include "dlaf/util_matrix.h"
+
+#include "pika/parallel/algorithms/for_each.hpp"
```

Comment:
```suggestion
#include "pika/algorithm.hpp"
```
We may provide a smaller header in the future just for `for_each` but `pika/parallel/algorithms/for_each.hpp` will at least not be the final location for that header, if it appears.